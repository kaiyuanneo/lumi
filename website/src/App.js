import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import RootContainer from './containers/RootContainer';
import rootReducer from './reducers';


const App = () => (
  <Provider store={createStore(rootReducer)}>
    <RootContainer />
  </Provider>
);

export default App;
