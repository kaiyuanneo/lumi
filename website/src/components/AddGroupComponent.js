import PropTypes from 'prop-types';
import React, { Component } from 'react';

import GroupCreateContainer from '../containers/GroupCreateContainer';
import GroupJoinContainer from '../containers/GroupJoinContainer';
import * as constants from '../static/constants';


class NewUserComponent extends Component {
  componentDidMount() {
    this.props.getUserFirstName();
  }
  render() {
    if (!this.props.firstName) {
      return null;
    }
    return (
      <div>
        <h1>{`${constants.NEW_USER_PAGE_TITLE}${this.props.firstName}!`}</h1>
        {constants.NEW_USER_PAGE_SUBTITLE}
        <GroupCreateContainer />
        <br />
        Or
        <br />
        <GroupJoinContainer />
      </div>
    );
  }
}

NewUserComponent.propTypes = {
  firstName: PropTypes.string.isRequired,
  getUserFirstName: PropTypes.func.isRequired,
};

export default NewUserComponent;
