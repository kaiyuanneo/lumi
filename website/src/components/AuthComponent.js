import firebase from 'firebase';
import React from 'react';
import { FirebaseAuth } from 'react-firebaseui';
import * as rp from 'request-promise';

import * as constants from '../static/constants';
import lumiLogo from '../static/images/logo.png';

/**
 * After user signs in with Facebook credentials, collect user information from Facebook
 * and save it in Firebase under the user record
 */
const setUserInfo = async (currentUser, credential) => {
  // Get user info from FB Graph API
  const fields = [
    'id',
    'first_name',
    'last_name',
    'locale',
    'timezone',
    'gender',
    'email',
    'birthday',
  ];
  const userInfoRequestOptions = {
    uri: constants.URL_FACEBOOK_GRAPH_API_ME,
    qs: {
      access_token: credential.accessToken,
      fields: fields.join(),
    },
    json: true,
  };
  const userInfoParsedBody = await rp(userInfoRequestOptions);

  // Store user info in user record
  const db = firebase.database();
  const userRef = db.ref(`${constants.DB_PATH_USERS}/${currentUser.uid}`);
  await userRef.update({
    ...userInfoParsedBody,
    // Save parsedBody "id" param as "asid", and save Firebase UID as "uid"
    asid: userInfoParsedBody.id,
    uid: currentUser.uid,
    profile_pic: currentUser.photoURL,
  });

  // Get PSID from FB Graph API via Lumi webserver. Need to go through webserver
  // because we cannot expose FB app access token in client code.
  const psidRequestOptions = {
    uri: constants.URL_LUMI_PSID,
    qs: {
      asid: userInfoParsedBody.id,
    },
    json: true,
  };
  const { psid } = await rp(psidRequestOptions);
  // If the user does not have a PSID yet (i.e. hasn't chatted with Lumi Chat), return here
  // and the next time the user logs in with a PSID, it will get populated by the below code.
  if (!psid) {
    return false;
  }

  // Store PSID in user record
  await userRef.update({
    psid,
  });

  // Create entry in user-psid-to-uid path so that Lumi can look up user info with PSID
  await db.ref(constants.DB_PATH_USER_PSID_TO_UID).update({
    [psid]: currentUser.uid,
  });

  // Avoid redirects after sign in
  return false;
};


const AuthComponent = () => {
  const uiConfig = {
    callbacks: {
      signInSuccess: setUserInfo,
    },
    signInFlow: 'popup',
    signInOptions: [firebase.auth.FacebookAuthProvider.PROVIDER_ID],
  };

  return (
    <div>
      <header className="App-header">
        <img src={lumiLogo} className="App-logo" alt="logo" />
        <h1 className="App-title">{constants.WEBSITE_TITLE}</h1>
      </header>
      <FirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
    </div>
  );
};

export default AuthComponent;
