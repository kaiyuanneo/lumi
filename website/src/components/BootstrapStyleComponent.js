import React from 'react';

import * as constants from '../static/constants';


const BootstrapStyleComponent = () => (
  <link rel="stylesheet" href={constants.URL_BOOTSTRAP_CSS} />
);

export default BootstrapStyleComponent;
