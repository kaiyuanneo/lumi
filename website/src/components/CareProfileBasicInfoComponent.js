import React from 'react';

import CareProfileEditWrapperContainer from '../containers/CareProfileEditWrapperContainer';
import * as constants from '../static/constants';
import * as utils from '../utils';


const CareProfileBasicInfoComponent = () => utils.wrapWithCareProfileTable((
  <tbody>
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_FIRST_NAME}
      title={constants.CARE_PROFILE_FIELD_TITLE_FIRST_NAME}
      formFieldGenerator={utils.getFirstNameFieldGenerator()}
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_LAST_NAME}
      title={constants.CARE_PROFILE_FIELD_TITLE_LAST_NAME}
      formFieldGenerator={utils.getLastNameFieldGenerator()}
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_BIRTHDAY}
      title={constants.CARE_PROFILE_FIELD_TITLE_BIRTHDAY}
      formFieldGenerator={utils.getBirthdayFieldGenerator()}
      isDateField
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_GENDER}
      title={constants.CARE_PROFILE_FIELD_TITLE_GENDER}
      formFieldGenerator={utils.getGenderFieldGenerator()}
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_EMAIL}
      title={constants.CARE_PROFILE_FIELD_TITLE_EMAIL}
      formFieldGenerator={utils.getEmailFieldGenerator()}
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_ADDRESS}
      title={constants.CARE_PROFILE_FIELD_TITLE_ADDRESS}
      formFieldGenerator={utils.getAddressFieldGenerator()}
    />
  </tbody>
));

export default CareProfileBasicInfoComponent;
