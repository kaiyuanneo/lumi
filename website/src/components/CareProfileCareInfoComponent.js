import React from 'react';

import CareProfileEditWrapperContainer from '../containers/CareProfileEditWrapperContainer';
import * as constants from '../static/constants';
import * as utils from '../utils';


const CareProfileCareInfoComponent = () => (
  <tbody>
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_NEEDS_AND_PREFERENCES}
      title={constants.CARE_PROFILE_FIELD_TITLE_NEEDS_AND_PREFERENCES}
      formFieldGenerator={utils.getNeedsAndPreferencesFieldGenerator()}
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_THINGS_THAT_DELIGHT}
      title={constants.CARE_PROFILE_FIELD_TITLE_THINGS_THAT_DELIGHT}
      formFieldGenerator={utils.getThingsThatDelightFieldGenerator()}
    />
    <CareProfileEditWrapperContainer
      fieldId={constants.CARE_PROFILE_FIELD_ID_PLACES_OF_INTEREST}
      title={constants.CARE_PROFILE_FIELD_TITLE_PLACES_OF_INTEREST}
      formFieldGenerator={utils.getPlacesOfInterestFieldGenerator()}
    />
  </tbody>
);

export default CareProfileCareInfoComponent;
