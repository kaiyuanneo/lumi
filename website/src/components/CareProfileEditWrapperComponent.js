import Flexbox from 'flexbox-react';
import PropTypes from 'prop-types';
import React from 'react';
import { Button, ButtonToolbar, Glyphicon } from 'react-bootstrap';

import * as constants from '../static/constants';


/**
 * NB: Need to pass isDateField prop to all date fields so that CareProfileEditWrapperComponent
 * calls the relevant handleChange function
 */
const CareProfileEditWrapperComponent = (props) => {
  // If edit mode is on, render the form field
  if (props.isInEditMode) {
    return (
      <tr>
        <td>{props.title}</td>
        <td>{props.formField}</td>
        <td>
          <Flexbox>
            <ButtonToolbar>
              <Button
                bsStyle="primary"
                disabled={props.saveButtonDisabled}
                onClick={props.saveFieldValueToDb}
              >
                {constants.BUTTON_TEXT_SAVE}
              </Button>
              <Button onClick={props.cancelEdits}>
                {constants.BUTTON_TEXT_CANCEL}
              </Button>
            </ButtonToolbar>
          </Flexbox>
        </td>
      </tr>
    );
  }
  // Otherwise, render the field value
  return (
    <tr>
      <td>{props.title}</td>
      <td className="multiline">{props.displayFieldValue}</td>
      <td>
        <Button onClick={props.enterEditMode}>
          <Glyphicon className="button-icon" glyph="pencil" />
          {constants.BUTTON_TEXT_EDIT}
        </Button>
      </td>
    </tr>
  );
};

CareProfileEditWrapperComponent.propTypes = {
  // Flag to display form field or display field value
  isInEditMode: PropTypes.bool.isRequired,

  // Display elements
  title: PropTypes.string.isRequired,
  formField: PropTypes.element.isRequired,
  saveButtonDisabled: PropTypes.bool.isRequired,
  displayFieldValue: PropTypes.string.isRequired,

  // Functions to control state
  enterEditMode: PropTypes.func.isRequired,
  cancelEdits: PropTypes.func.isRequired,
  saveFieldValueToDb: PropTypes.func.isRequired,
};

export default CareProfileEditWrapperComponent;
