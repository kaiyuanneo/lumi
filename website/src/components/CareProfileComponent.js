import Flexbox from 'flexbox-react';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Image, Panel, PanelGroup } from 'react-bootstrap';
import DropDownIcon from 'react-icons/lib/md/arrow-drop-down';

import CareProfileBasicInfoComponent from '../components/CareProfileBasicInfoComponent';
import CareProfileMedicalInfoComponent from '../components/CareProfileMedicalInfoComponent';
import CareProfileCareInfoComponent from '../components/CareProfileCareInfoComponent';
import CareProfileSelectCareRecipientContainer from '../containers/CareProfileSelectCareRecipientContainer';
import * as constants from '../static/constants';


class CareProfileComponent extends Component {
  componentDidMount() {
    this.props.getCareRecipient();
  }

  render() {
    if (!this.props.fetched) {
      return null;
    }
    // Render care recipient selector if this group has no care recipient yet
    if (!this.props.uid) {
      return <CareProfileSelectCareRecipientContainer />;
    }
    return (
      <Flexbox flexDirection="column">
        <Flexbox flexDirection="column">
          <Flexbox alignSelf="center">
            <Image src={this.props.profilePic} circle responsive />
          </Flexbox>
          <h4>Care Profile</h4>
        </Flexbox>
        <br />
        <Flexbox flexDirection="column" alignContent="center">
          <PanelGroup accordion id="care-profile">
            <Panel eventKey="1">
              <Panel.Heading>
                <Panel.Title toggle>
                  Basic Info
                  <DropDownIcon size={constants.CARE_PROFILE_ICON_SIZE} />
                </Panel.Title>
              </Panel.Heading>
              <Panel.Body collapsible>
                <CareProfileBasicInfoComponent />
              </Panel.Body>
            </Panel>
            <Panel eventKey="2">
              <Panel.Heading>
                <Panel.Title toggle>
                  Medical Info
                  <DropDownIcon size={constants.CARE_PROFILE_ICON_SIZE} />
                </Panel.Title>
              </Panel.Heading>
              <Panel.Body collapsible>
                <CareProfileMedicalInfoComponent />
              </Panel.Body>
            </Panel>
            <Panel eventKey="3">
              <Panel.Heading>
                <Panel.Title toggle>
                  Care Info
                  <DropDownIcon size={constants.CARE_PROFILE_ICON_SIZE} />
                </Panel.Title>
              </Panel.Heading>
              <Panel.Body collapsible>
                <CareProfileCareInfoComponent />
              </Panel.Body>
            </Panel>
          </PanelGroup>
        </Flexbox>
      </Flexbox>
    );
  }
}

CareProfileComponent.propTypes = {
  // Props computed from Redux state
  fetched: PropTypes.bool.isRequired,
  uid: PropTypes.string.isRequired,
  profilePic: PropTypes.string.isRequired,
  // Props that call dispatch actions
  getCareRecipient: PropTypes.func.isRequired,
};

export default CareProfileComponent;
