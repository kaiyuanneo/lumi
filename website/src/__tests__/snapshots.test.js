import * as firebase from 'firebase';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import sinon from 'sinon';

import App from '../App';
import AddGroupComponent from '../components/AddGroupComponent';
import AuthComponent from '../components/AuthComponent';
import BootstrapStyleComponent from '../components/BootstrapStyleComponent';
import CareProfileBasicInfoComponent from '../components/CareProfileBasicInfoComponent';
import CareProfileCareInfoComponent from '../components/CareProfileCareInfoComponent';
import CareProfileComponent from '../components/CareProfileComponent';
import CareProfileEditWrapperComponent from '../components/CareProfileEditWrapperComponent';
import CareProfileMedicalInfoComponent from '../components/CareProfileMedicalInfoComponent';
import CareProfileNewMemberComponent from '../components/CareProfileNewMemberComponent';
import CareProfileNewMemberFormFieldComponent from '../components/CareProfileNewMemberFormFieldComponent';
import CareProfileSelectCareRecipientComponent from '../components/CareProfileSelectCareRecipientComponent';
import GroupCreateComponent from '../components/GroupCreateComponent';
import GroupJoinComponent from '../components/GroupJoinComponent';
import HomeComponent from '../components/HomeComponent';
import NavBottomComponent from '../components/NavBottomComponent';
import NavTopComponent from '../components/NavTopComponent';
import RootComponent from '../components/RootComponent';
import TimelineComponent from '../components/TimelineComponent';
import TimelineStoryHeaderComponent from '../components/TimelineStoryHeaderComponent';
import TimelineStoryContentComponent from '../components/TimelineStoryContentComponent';
import TimelineStoryFooterComponent from '../components/TimelineStoryFooterComponent';


describe('App', () => {
  let authStub;
  beforeAll(() => {
    authStub = sinon.stub(firebase, 'auth').returns({
      onAuthStateChanged: sinon.stub(),
    });
  });

  afterAll(() => {
    authStub.restore();
  });

  it('App renders', () => {
    const component = TestRenderer.create(<App />).toTree();
    expect(component).toBeTruthy();
  });
});


it('AddGroupComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render((
    <AddGroupComponent
      firstName=""
      getUserFirstName={() => {}}
    />
  ));
  expect(component).toMatchSnapshot();
});


it('AuthComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render((
    <AuthComponent
      firebaseAuth={() => {}}
      uiConfig={{}}
    />
  ));
  expect(component).toMatchSnapshot();
});


it('BootstrapStyleComponent', () => {
  const component = TestRenderer.create(<BootstrapStyleComponent />).toJSON();
  expect(component).toMatchSnapshot();
});


it('CareProfileBasicInfoComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render(<CareProfileBasicInfoComponent />);
  expect(component).toMatchSnapshot();
});


it('CareProfileCareInfoComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render(<CareProfileCareInfoComponent />);
  expect(component).toMatchSnapshot();
});


it('CareProfileComponent', () => {
  const component = TestRenderer.create((
    <CareProfileComponent
      contentComponent={<div />}
      fetched={false}
      fullName=""
      getCareRecipient={() => {}}
      infoCategory=""
      profilePic=""
      saveCareProfileInfoCategory={() => {}}
      uid=""
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('CareProfileEditWrapperComponent', () => {
  const component = TestRenderer.create((
    <CareProfileEditWrapperComponent
      cancelEdits={() => {}}
      displayFieldValue=""
      enterEditMode={() => {}}
      formField={<div />}
      isInEditMode={false}
      saveButtonDisabled={false}
      saveFieldValueToDb={() => {}}
      title=""
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('CareProfileMedicalInfoComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render(<CareProfileMedicalInfoComponent />);
  expect(component).toMatchSnapshot();
});


it('CareProfileNewMemberComponent', () => {
  const component = TestRenderer.create((
    <CareProfileNewMemberComponent
      firstNameFormField={<div />}
      lastNameFormField={<div />}
      birthdayFormField={<div />}
      genderFormField={<div />}
      emailFormField={<div />}
      addressFormField={<div />}
      typeOfDementiaFormField={<div />}
      dateOfDiagnosisFormField={<div />}
      medicationsFormField={<div />}
      providersFormField={<div />}
      needsAndPreferencesFormField={<div />}
      thingsThatDelightFormField={<div />}
      placesOfInterestFormField={<div />}
      isSaveButtonDisabled={false}
      saveNewMember={() => {}}
      unmountFunc={() => {}}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('CareProfileNewMemberFormFieldComponent', () => {
  const component = TestRenderer.create((
    <CareProfileNewMemberFormFieldComponent
      fieldTitle=""
      formField={<div />}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('CareProfileSelectCareRecipientComponent', () => {
  const component = TestRenderer.create((
    <CareProfileSelectCareRecipientComponent
      fetchGroupMembers={() => {}}
      memberList={[]}
      selectedMember=""
      shouldRenderNewMemberForm={false}
      isSelectButtonDisabled={false}
      updateSelectedMember={() => {}}
      handleClickSelect={() => {}}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('GroupCreateComponent', () => {
  const component = TestRenderer.create((
    <GroupCreateComponent
      firstNameFieldValue=""
      lastNameFieldValue=""
      isCreateButtonDisabled={false}
      createGroup={() => {}}
      saveCareProfileFirstName={() => {}}
      saveCareProfileLastName={() => {}}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('GroupJoinComponent', () => {
  const component = TestRenderer.create((
    <GroupJoinComponent
      groupIdFieldValue=""
      isJoinButtonDisabled={false}
      joinGroup={() => {}}
      handleChange={() => {}}
      handle
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('HomeComponent', () => {
  const component = TestRenderer.create((
    <HomeComponent
      getIsAuthUserInGroup={() => {}}
      saveWindowWidth={() => {}}
      shouldComponentRender={false}
      productComponent={<div />}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('NavBottomComponent', () => {
  const component = TestRenderer.create((
    <NavBottomComponent
      activeKey=""
      switchProduct={() => {}}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('NavTopComponent', () => {
  const component = TestRenderer.create((
    <NavTopComponent
      groups={[]}
      disableCopyButton={false}
      getGroupInfo={() => {}}
      handleNavSelect={() => {}}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('RootComponent', () => {
  const component = TestRenderer.create((
    <RootComponent
      getIsSignedIn={() => {}}
    />
  )).toJSON();
  expect(component).toMatchSnapshot();
});


it('TimelineComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render((
    <TimelineComponent
      numMessages={0}
      sortedMessages={new Map()}
      syncMessages={() => {}}
      shouldRenderMessage={() => {}}
      windowWidth={0}
    />
  ));
  expect(component).toMatchSnapshot();
});

it('TimelineStoryHeaderComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render((
    <TimelineStoryHeaderComponent
      getDateString={() => {}}
      messageValue={{}}
    />
  ));
  expect(component).toMatchSnapshot();
});

it('TimelineStoryContentComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render((
    <TimelineStoryContentComponent
      messageValue={{ attachments: [{ payload: { url: '' } }] }}
    />
  ));
  expect(component).toMatchSnapshot();
});

it('TimelineStoryFooterComponent', () => {
  const renderer = new ShallowRenderer();
  const component = renderer.render((
    <TimelineStoryFooterComponent
      getCategoryName={() => {}}
      messageValue={{}}
    />
  ));
  expect(component).toMatchSnapshot();
});
