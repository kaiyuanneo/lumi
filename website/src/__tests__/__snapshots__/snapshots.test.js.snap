// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthComponent 1`] = `
<div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <h1
      className="App-title"
    >
      Lumi
    </h1>
  </header>
  <FirebaseAuth
    firebaseAuth={undefined}
    uiConfig={Object {}}
  />
</div>
`;

exports[`BootstrapStyleComponent 1`] = `
<link
  crossOrigin="anonymous"
  href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
  integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
  rel="stylesheet"
/>
`;

exports[`GroupCreateComponent 1`] = `
<div>
  <h2>
    Create New Group
  </h2>
  <form>
    <div
      className="form-group has-feedback"
    >
      <label
        className="control-label"
        htmlFor="formGroupNewGroup"
      >
        Choose a group name for your new group
      </label>
      <input
        className="form-control"
        id="formGroupNewGroup"
        onChange={[Function]}
        placeholder="Enter new group name"
        type="text"
        value=""
      />
      <span
        className="help-block"
      >
        Lumi will generate a Group ID to share with family and friends on the next page
      </span>
    </div>
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Create Group
    </button>
  </form>
</div>
`;

exports[`GroupJoinComponent 1`] = `
<div>
  <h2>
    Join Existing Group
  </h2>
  <form>
    <div
      className="form-group has-feedback"
    >
      <label
        className="control-label"
        htmlFor="formGroupJoinGroup"
      >
        Join an existing Lumi Group
      </label>
      <input
        className="form-control"
        id="formGroupJoinGroup"
        onChange={[Function]}
        placeholder="Enter Lumi Group ID"
        type="text"
        value=""
      />
      <span
        className="help-block"
      >
        This ID must be a valid group ID of an existing Lumi Group
      </span>
    </div>
    <button
      className="btn btn-lg btn-default btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Join Group
    </button>
  </form>
</div>
`;

exports[`HomeComponent 1`] = `null`;

exports[`NavBottomComponent 1`] = `
<nav
  className="nav-bottom navbar navbar-default navbar-fixed-bottom"
>
  <div
    className="container"
  >
    <div
      className="sc-bdVaJa cNkfwA"
    >
      <ul
        className="nav navbar-nav"
        role={null}
      >
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <div
              className="sc-bdVaJa djJVkl"
            >
              <span
                className="glyphicon glyphicon-home"
              />
              Timeline
            </div>
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <div
              className="sc-bdVaJa djJVkl"
            >
              <span
                className="glyphicon glyphicon-plus"
              />
              Chat
            </div>
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <div
              className="sc-bdVaJa djJVkl"
            >
              <span
                className="glyphicon glyphicon-info-sign"
              />
              Summary
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`NavTopComponent 1`] = `
<div
  className="nav-top"
>
  <nav
    className="navbar navbar-default navbar-fixed-top"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <b>
            Lumi
          </b>
        </a>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <ul
          className="nav navbar-nav navbar-right"
          role={null}
        >
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Sign Out
            </a>
          </li>
        </ul>
        <p
          className="navbar-text navbar-right"
        />
      </div>
    </div>
  </nav>
</div>
`;

exports[`NewUserComponent 1`] = `
<div>
  <h1>
    Welcome to Lumi, !
  </h1>
  Before we begin, you must either create or join a Lumi Group.
  <Connect(GroupCreateComponent) />
  <br />
  Or
  <br />
  <Connect(GroupJoinComponent) />
</div>
`;

exports[`RootComponent 1`] = `null`;

exports[`SummaryBasicInfoComponent 1`] = `
<Table
  bordered={true}
  bsClass="table"
  condensed={false}
  hover={true}
  responsive={false}
  striped={false}
>
  <thead>
    <tr>
      <th
        className="product-table-header"
      >
        Field
      </th>
      <th
        className="product-table-header"
      >
        Value
      </th>
      <th
        className="product-table-header"
      >
        Options
      </th>
    </tr>
  </thead>
  <tbody>
    <Connect(SummaryEditWrapperComponent)
      fieldId="firstName"
      formFieldGenerator={[Function]}
      title="First Name"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="lastName"
      formFieldGenerator={[Function]}
      title="Last Name"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="birthday"
      formFieldGenerator={[Function]}
      isDateField={true}
      title="Birthday"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="gender"
      formFieldGenerator={[Function]}
      title="Gender"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="email"
      formFieldGenerator={[Function]}
      title="Email"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="address"
      formFieldGenerator={[Function]}
      title="Address"
    />
  </tbody>
</Table>
`;

exports[`SummaryCareInfoComponent 1`] = `
<Table
  bordered={true}
  bsClass="table"
  condensed={false}
  hover={true}
  responsive={false}
  striped={false}
>
  <thead>
    <tr>
      <th
        className="product-table-header"
      >
        Field
      </th>
      <th
        className="product-table-header"
      >
        Value
      </th>
      <th
        className="product-table-header"
      >
        Options
      </th>
    </tr>
  </thead>
  <tbody>
    <Connect(SummaryEditWrapperComponent)
      fieldId="needsAndPreferences"
      formFieldGenerator={[Function]}
      title="Needs and Preferences"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="thingsThatDelight"
      formFieldGenerator={[Function]}
      title="Things that Delight"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="placesOfInterest"
      formFieldGenerator={[Function]}
      title="Places of Interest"
    />
  </tbody>
</Table>
`;

exports[`SummaryComponent 1`] = `null`;

exports[`SummaryEditWrapperComponent 1`] = `
<tr>
  <td>
    
  </td>
  <td
    className="multiline"
  >
    
  </td>
  <td>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="button-icon glyphicon glyphicon-pencil"
      />
      Edit
    </button>
  </td>
</tr>
`;

exports[`SummaryMedicalInfoComponent 1`] = `
<Table
  bordered={true}
  bsClass="table"
  condensed={false}
  hover={true}
  responsive={false}
  striped={false}
>
  <thead>
    <tr>
      <th
        className="product-table-header"
      >
        Field
      </th>
      <th
        className="product-table-header"
      >
        Value
      </th>
      <th
        className="product-table-header"
      >
        Options
      </th>
    </tr>
  </thead>
  <tbody>
    <Connect(SummaryEditWrapperComponent)
      fieldId="typeOfDementia"
      formFieldGenerator={[Function]}
      title="Type of Dementia"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="dateOfDiagnosis"
      formFieldGenerator={[Function]}
      isDateField={true}
      title="Date of Diagnosis"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="medications"
      formFieldGenerator={[Function]}
      title="Medications"
    />
    <Connect(SummaryEditWrapperComponent)
      fieldId="providers"
      formFieldGenerator={[Function]}
      title="Providers"
    />
  </tbody>
</Table>
`;

exports[`SummaryNewMemberComponent 1`] = `
<div
  className="sc-bdVaJa djJVkl"
>
  <h2>
    Please share details about the new member :)
  </h2>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          First Name
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Last Name
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Birthday
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Gender
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Email
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Address
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Type of Dementia
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Date of Diagnosis
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Medications
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Providers
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Needs and Preferences
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Things that Delight
        </td>
        <td>
          <div />
        </td>
      </tr>
      <tr>
        <td>
          Places of Interest
        </td>
        <td>
          <div />
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="sc-bdVaJa cXgeQK"
  >
    <div
      className="btn-toolbar"
      role="toolbar"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`SummaryNewMemberFormFieldComponent 1`] = `
<tr>
  <td>
    
  </td>
  <td>
    <div />
  </td>
</tr>
`;

exports[`SummarySelectCareRecipientComponent 1`] = `
<div
  className="sc-bdVaJa hgkmdP"
>
  <h2>
    Welcome to Summary! Please choose a care recipient:
  </h2>
  <div
    className="form-group"
  >
    <select
      className="form-control"
      id="userList"
      onChange={[Function]}
      placeholder="Select a care recipient"
      type={undefined}
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Select
  </button>
</div>
`;

exports[`TimelineComponent 1`] = `
<div>
  <Connect(TimelineFiltersComponent) />
  <Table
    bordered={false}
    bsClass="table"
    condensed={false}
    hover={false}
    responsive={false}
    striped={false}
  >
    <tbody>
      
    </tbody>
  </Table>
</div>
`;
